### ISSUE TITLE
Tests fail

### ISSUE THREAD

**Issue by dunetna** [tristanheaven/gtkhash#9]
Hi,
If I try to run tests I get errors. I do the following steps:
Build gtkhash:
```
$ ./configure --enable-mhash  # no error
$ make  # no error
```
If I run gtkhash, everything seems to be ok.
But when I run the tests, all of them fail.

First, I try to run tests from Makefile and get a Segmentation fault:
```
$ make check # error, with a segmentation fault
```
logs: [make-check.txt](https://github.com/tristanheaven/gtkhash/files/710989/make-check.txt), [gtkhash-test.txt](https://github.com/tristanheaven/gtkhash/files/710991/gtkhash-test.txt)

If I run tests from gtkhash_test I also have errors:
```
$ src/gtkhash-test # errors of disabled functions
```
log: [gtkhash-test-run.txt](https://github.com/tristanheaven/gtkhash/files/710993/gtkhash-test-run.txt)

In this case, all test functions fail with the same type of error:
```
/hash/func/MD2: **
GtkHash:ERROR:test.c:53:select_func: 'hash.funcs[id].enabled == active' should be TRUE
**
GtkHash:ERROR:test.c:74:test_hash_func_digest: assertion failed (output == digest): ("" == "8350e5a3e24c153df2275c9f80692773")
**
GtkHash:ERROR:test.c:53:select_func: 'hash.funcs[id].enabled == active' should be TRUE
**
GtkHash:ERROR:test.c:74:test_hash_func_digest: assertion failed (output == digest): ("" == "03d85a0d629d2c442e987525319fc471")
FAIL
```

I don't know if I am doing something wrong. Could you give me a hint?
I need to run tests successfully to build the Debian package of gtkhash.

Thanks a lot.

**Comment by tristanheaven** [comment]
Seems something doesn't like the default screen resolution/depth xvfb-run uses. I'm not sure what's changed that broke this. Maybe gtk.

You could try:

```xvfb-run -s "-screen 0 800x600x24" src/gtkhash-test```

which seems to work.

Btw, I'd recommend using gcrypt or nettle if possible. I don't think mhash is maintained these days, and you'll get SHA3 using either of those instead.

**Comment by dunetna** [comment]
Sometimes it still fails (I don't know why sometimes fails and sometimes not), even with the patch. Here you have the Debian bug with gtkhash build log: https://bugs.debian.org/852893
Could you reopen the bug?

**Comment by tristanheaven** [comment]
Should be fixed with b8813894960dc, thanks.
